### PR TITLE
To index function

### PR DESCRIPTION
**Pull request by jobo322**
add toIndex function and  testCase

### CODE REVIEW

File: src/range/Ranges.js [hunk 1]
```diff
@@ -204,6 +214,109 @@ class Ranges extends Array {
...
     }
+    /**
+     * Return an array of deltas and multiplicity for an index database
+     * @returns {Array} [{delta, multiplicity},...]
+     */
+    toIndex() {
+        var index = [];
+        this.compactPatterns();
+        for(var range of this) {
+            if (range.signal == undefined || range.signal == []) {
```
Comment (maasencioh) [inline src/range/Ranges.js]: consider using `===` instead of `==`, another way is to write this as `!range.signal`

Comment (targos) [reply src/range/Ranges.js]: please don't write `!range.signal`

Comment (maasencioh) [reply src/range/Ranges.js]: why not?

Comment (targos) [reply src/range/Ranges.js]: It's better to be explicit. BTW it's impossible for `range.signal == []` to be true (`[]` creates a new array object. it cannot be equal to an existing value)

**Pull request closed without merging**